src/flint/test/test_arb.py

```diff
@@ -0,0 +1,354 @@
+    assert true_interval in actual
+
+def test_arb_abs():
+    """`arb.__abs__` works as expected."""
+    arb_val = arb(1, 2)  # [-1,3]
+    actual = abs(arb_val)
+    true_interval = arb(1.5, 1.5)
+    assert true_interval in actual
+
+def test_arb_neg():
+    """`arb.neg` works as expected."""
+    arb_val = arb(1, 2)  # [-1,3]
+    actual = arb_val.neg(exact=True)
+    true_interval = arb(-2, 1)  # [-3,1]
+    assert true_interval in actual
+
+def test_arb_neg_dunder():
+    """`arb.__neg__` works as expected."""
+    arb_val = arb(1, 2)  # [-1,3]
+    actual = -arb_val
+    true_interval = arb(-2, 1)  # [-3,1]
+    assert true_interval in actual
+
+def test_arb_sgn():
+    """`arb.sgn` works as expected."""
+    arb1 = arb(1, 0.5)  # [0.5,1.5]
+    arb2 = arb(-1, 0.5)  # [-1.5,-0.5]
+    arb3 = arb(1, 2)  # [-1,3]
+    assert_almost_equal(float(arb1.sgn()), 1)
+    assert_almost_equal(float(arb2.sgn()), -1)
+    # arb3 contains both positive and negative numbers
+    # So, arb_sgn returns [0, 1]
+    assert_almost_equal(float(arb3.sgn().mid()), 0)
+    assert_almost_equal(float(arb3.sgn().rad()), 1)
+
+def test_arb_erfinv():
+    """`arb.erfinv` works as expected."""
+    midpoint = (math.erf(1 / 8) + math.erf(1 / 16)) / 2
+    radius = midpoint - math.erf(1 / 16)
+    arb_val = arb(midpoint, radius)
+    with ctx.workprec(100):
+        actual = arb_val.erfinv()
+    true_interval = arb(3 / 32, 1 / 32)  # [1/16, 1/8]
+    assert true_interval in actual
+
+def test_arb_erf():
+    """`arb.erf` works as expected."""
+    arb_val = arb(2, 1)
+    with ctx.workprec(100):
+        actual = arb_val.erf()
+    true_interval = arb(
+        (math.erf(1) + math.erf(3)) / 2,
+        (math.erf(1) + math.erf(3)) / 2 - math.erf(1)
+    )
+    assert true_interval in actual
+
+def test_arb_erfc():
+    """`arb.erfc` works as expected."""
+    arb_val = arb(2, 1)
+    with ctx.workprec(100):
+        actual = arb_val.erfc()
+    true_interval = arb(
+        (math.erfc(1) + math.erfc(3)) / 2,
+        (math.erfc(1) + math.erfc(3)) / 2 - math.erfc(3)
+    )
+    assert true_interval in actual
+
+def test_arb_const_pi():
+    """`arb.pi` works as expected."""
+    with ctx.workprec(100):
+        actual = arb.pi()
+    interval_around_pi = arb(math.pi, 1e-10)
+    assert actual in interval_around_pi
+
+def test_arb_union():
+    """`arb.union` works as expected."""
+    arb1 = arb(1, 0.5)  # [0.5,1.5]
+    arb2 = arb(3, 0.5)  # [2.5,3.5]
+    with ctx.workprec(100):
+        actual = arb1.union(arb2)
+    true_interval = arb(2, 1.5)  # [0.5, 3.5]
+    assert true_interval in actual
+
+def test_arb_sum():
+    """`arb.__sum__` works as expected."""
+    arb1 = arb(1, 0.5)  # [0.5,1.5]
+    arb2 = arb(2, 0.5)  # [1.5,2.5]
+    arb3 = arb(3, 0.5)  # [2.5,3.5]
+    with ctx.workprec(100):
+        actual = arb1 + arb2 + arb3
+    true_interval = arb(6, 1.5)  # [4.5, 7.5]
+    assert true_interval in actual
+
+all_tests = [
```

Comment:
There should be a test that all the tests in this file are in this list or otherwise someone will end up adding tests here without adding them to the list. There is a test function that does this in `test_all.py` which could be adapted to do check for this file as well.

Reply:
We could require pytest but I don't think it would particularly make writing tests easier. Either way you mostly just write a lot of functions called `test_foo`. The potential advantages of pytest are things like selecting particular tests or running tests in parallel but the test suite is fast enough not to need those. One thing pytest is not good at is doctests which make up a good fraction of the test suite.